File: 0134-gas-station/0134-gas-station.java
class Solution {
    public int canCompleteCircuit(int[] gas, int[] cost) {
        int gasInTank = 0;
        int n=cost.length;
        int start=0;
        int sum=0;
        for(int i=0;i<n;i++){
            gasInTank += gas[i] - cost[i];
            sum += gas[i] - cost[i];
            if(gasInTank<0){
                start = i+1;
                gasInTank = 0;
            }
        }
        return sum >=0?start : -1;
    }
}

File: 0134-gas-station/0134-gas-station.py
class Solution:
    def canCompleteCircuit(self, gas: List[int], cost: List[int]) -> int:
        total_tank = 0
        current_tank = 0
        start = 0
        for i in range(len(gas)):
            diff = gas[i] - cost[i]
            total_tank += diff
            current_tank += diff
            
            if current_tank < 0:
                start = i + 1
                current_tank = 0
        return start if total_tank >= 0 else -1